### ISSUE TITLE
Wifi (BCM 4352) working in ABG mode only

### ISSUE THREAD

**Issue by gaborkr** [advancingu/XPS13Linux#10]
Using the proprietary Broadcom driver (bcmwl-kernel-source 6.30.223.248) on Ubuntu 15.04 / kernel 3.19 the connection to an N-class router is ABG only:

```
$ iwconfig
wlan0 IEEE 802.11abg ESSID:"NETGEAR93" 
Mode:Managed Frequency:2.412 GHz Access Point: 20:0C:C8:29:B8:22 
Retry short limit:7 RTS thr:off Fragment thr:off
Power Management:on
```

Signal also seems to be weaker than other wifi devices.

**Comment by gaborkr** [comment]
A "workaround" is to buy an Intel 7265 module and replace the Broadcom card.
[This bug](https://bugs.launchpad.net/ubuntu/+source/bcmwl/+bug/1451789) seems to be related.

**Comment by dchambers** [comment]
@butabeka, do you definitely know that the Intel WiFi card works? For example, does it always work when coming out of a suspend, and can it connect to Wireless N hubs without issue?

**Comment by gaborkr** [comment]
Before posting this comment I closed the lid and reopened it (I'm running on battery now), and the 7265 wifi resumed without a problem. So it works for me, at least for now. That is all I can say. Some people report problems with kernel 4.0 / iwlwifi, just FYI. I'm on 3.19, everything seems to be stable so far.

**Comment by dchambers** [comment]
@butabeka, yep, my current laptop has an Intel wifi adaptor, and it doesn't always come out of suspend reliably and doesn't always get on with Wireless N routers, hence my interest. I really do hope it continues to work reliably for you, because I really do want a reliable Linux laptop!
